libopenage/input/input_manager.cpp

```diff
@@ -16,84 +17,72 @@ InputManager::InputManager()
-	keys.emplace(sdl_mouse(1), action_manager.get("PAINT_TERRAIN"));
-	keys.emplace(sdl_mouse(1), action_manager.get("SELECT"));
-	keys.emplace(sdl_mouse(3), action_manager.get("DESELECT"));
-	keys.emplace(sdl_wheel(1), action_manager.get("FORWARD"));
-	keys.emplace(sdl_wheel(-1), action_manager.get("BACK"));
+	this->set_bind("`", "TOGGLE_CONSOLE");
+	this->set_bind("Return", "START_GAME");
+	this->set_bind("Escape", "STOP_GAME");
+	this->set_bind("F1", "TOGGLE_HUD");
+	this->set_bind("F2", "SCREENSHOT");
+	this->set_bind("F3", "TOGGLE_DEBUG_OVERLAY");
+	this->set_bind("F4", "TOGGLE_DEBUG_GRID");
+	this->set_bind("F5", "QUICK_SAVE");
+	this->set_bind("F9", "QUICK_LOAD");
+	this->set_bind("Space", "TOGGLE_BLENDING");
+	this->set_bind("F12", "TOGGLE_PROFILER");
+	this->set_bind("m", "TOGGLE_CONSTRUCT_MODE");
+	this->set_bind("p", "TOGGLE_UNIT_DEBUG");
+	this->set_bind("t", "TRAIN_OBJECT");
+	this->set_bind("y", "ENABLE_BUILDING_PLACEMENT");
+	this->set_bind("Ctrl z", "DISABLE_SET_ABILITY");
+	this->set_bind("Ctrl x", "SET_ABILITY_MOVE");
+	this->set_bind("Ctrl c", "SET_ABILITY_GATHER");
+	this->set_bind("g", "SET_ABILITY_GARRISON");
+	this->set_bind("v", "SPAWN_VILLAGER");
+	this->set_bind("Delete", "KILL_UNIT");
+	this->set_bind("q", "BUILDING_1");
+	this->set_bind("w", "BUILDING_2");
+	this->set_bind("e", "BUILDING_3");
+	this->set_bind("r", "BUILDING_4");
+	this->set_bind("Ctrl q", "BUILDING_5");
+	this->set_bind("Ctrl w", "BUILDING_6");
+	this->set_bind("Ctrl e", "BUILDING_7");
+	this->set_bind("Ctrl r", "BUILDING_8");
+	this->set_bind("z", "BUILDING_TOWN_CENTER");
+	this->set_bind("1", "SWITCH_TO_PLAYER_1");
+	this->set_bind("2", "SWITCH_TO_PLAYER_2");
+	this->set_bind("3", "SWITCH_TO_PLAYER_3");
+	this->set_bind("4", "SWITCH_TO_PLAYER_4");
+	this->set_bind("5", "SWITCH_TO_PLAYER_5");
+	this->set_bind("6", "SWITCH_TO_PLAYER_6");
+	this->set_bind("7", "SWITCH_TO_PLAYER_7");
+	this->set_bind("8", "SWITCH_TO_PLAYER_8");
+	this->set_bind("Up", "UP_ARROW");
+	this->set_bind("Down", "DOWN_ARROW");
+	this->set_bind("Left", "LEFT_ARROW");
+	this->set_bind("Right", "RIGHT_ARROW");
+	this->set_bind("MOUSE 1", "PAINT_TERRAIN");
+	this->set_bind("WHEEL 1", "FORWARD");
+	this->set_bind("WHEEL -1", "BACK");
 }
 
-std::string InputManager::get_key(const std::string action_str) {
+std::string InputManager::get_bind(const std::string &action_str) {
 	ActionManager &action_manager = Engine::get().get_action_manager();
 
 	for (auto &it : this->keys) {
 		if (action_manager.is(action_str, it.second)) {
-			std::string key_str = std::string(SDL_GetKeyName(it.first.cc.code));
-
-			auto end = it.first.mod.end();
-			if (it.first.mod.find(modifier::ALT) != end) {
-				key_str = "Alt "+key_str;
+			switch (it.first.cc.eclass) {
+				case event_class::MOUSE_BUTTON :
+					return this->mouse_bind_to_string(it.first);
+				case event_class::MOUSE_WHEEL :
+					return this->wheel_bind_to_string(it.first);
+				default:
+					return this->key_bind_to_string(it.first);
```

Comment:
Please dedent this region by one tab. The case labels are at the same indent level as the switch, but the code is then indented one tab.